libs/entitlement/src/main/java/org/elasticsearch/entitlement/runtime/policy/PathLookup.java

```diff
@@ -10,5 +10,13 @@
 package org.elasticsearch.entitlement.runtime.policy;
 
 import java.nio.file.Path;
+import java.util.function.Function;
+import java.util.stream.Stream;
 
-public record PathLookup(Path configDir, Path[] dataDirs, Path tempDir) {}
+public record PathLookup(
+    Path configDir,
+    Path[] dataDirs,
+    Path tempDir,
+    Function<String, String> settingResolver,
```

Comment:
This is why I think and interface is better than a record here; a record with 2 Function is just and interface in disguise. OK to keep *Dir() as simple getters though.
Not blocking.

Reply:
I'm still in favor of the record here as there is only a single implementation. The resolvers are just necessary due to `Settings` not being available in the entitlements lib

Reply:
I agree with Moritz, I prefer a record because it is "things" we are passing in. Creating an interface would require creating an implementation, but the record provides that.